### DISCUSSION TITLE
about clickhouse recover data

### DISCUSSION THREAD

**Discussion by OrionPotter** [ClickHouse/clickhouse-kafka-connect#248] [Q&A] [answered]
During the data recovery test, I have a centralized authority of four machines, 2S2R architecture, deploy a single kepper separately, suppose one of the machines is down and cannot be recovered, I add a new server, and copy metadata and data from the replica to the new one, and also create a forcerestoredata on the new machine, there is some data loss, can you give a tutorial on data recovery?

**Comment by Paultagoras** [comment]
Hi @BJPotter - I guess it depends on what you mean. Where are you seeing the data loss?

There's two ends to the data flow: A) The Kafka side of things (with Zookeeper or KRaft) and B) The ClickHouse db side of things. Are you seeing data missing from the kafka topic after a failure? Is it missing data from a ClickHouse table?

**Reply by OrionPotter** [reply]
I inserted 30 pieces of data into the distributed table, then deleted the data directory of the clickhouse of cluster1, and then inserted 30 pieces of data into the distributed table on cluster2, and then restored cluster1, at this time, cluster1 lost part of the 30 pieces of data inserted at the beginning.

**Reply by Paultagoras** [reply, marked as answer]
AH I see now - the best place to get an answer for that would be either at [our ClickHouse DB repo](https://github.com/ClickHouse/ClickHouse) or on the [community Slack server](https://clickhouse.com/slack) - this repo is meant to handle the connector between ClickHouse and Apache Kafka. Hope that clears things up!

**Reply by OrionPotter** [reply]
thank you

**Reply by Paultagoras** [reply]
No worries :)
